src/dmtcpworker.cpp

```diff
@@ -464,17 +464,27 @@ void
 DmtcpWorker::postCheckpoint()
 {
   WorkerState::setCurrentState(WorkerState::CHECKPOINTED);
+
+  // TODO: Merge this barrier with the previous `sendCkptFilename` msg.
+  JTRACE("Waiting for Write-Ckpt barrier");
```

Comment:
Please add a comment here.  Perhaps something like:
```
// It's important for all processes to rename .dmtcp.temp to .temp at the same time.
// If one process is gigabytes-large, then without the barrier, it would rename long after
// other renaming has finished, leaving the checkpoints in an inconsistent state for many minutes.
```